Build fails with strict-aliasing violations

I tried to build with the following *FLAGS to optimize the build: `-flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing`

The -Werror=* flags are important to detect cases where the compiler can try to optimize based on assuming Undefined Behavior (UB) cannot happen, and miscompile code that has UB in it. strict-aliasing issues are always bad but LTO can make them even worse.

I got this error:
```
FAILED: [code=1] CMakeFiles/libh2o.dir/lib/common/multithread.c.o 
/usr/bin/x86_64-pc-linux-gnu-gcc -DH2O_HAS_GITREV_H -DLIBCAP_FOUND -DLIBC_HAS_BACKTRACE -Dlibh2o_EXPORTS -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/include -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/cloexec -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/brotli/c/include -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/golombset -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/hiredis -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/libgkc -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/libyrmcds -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/klib -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/neverbleed -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/picohttpparser -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/picotest -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/picotls/deps/cifra/src/ext -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/picotls/deps/cifra/src -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/picotls/deps/micro-ecc -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/picotls/include -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/quicly/include -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/yaml/include -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/deps/yoml -I/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999_build  -std=c99 -g3 -Wall -Wno-unused-value -Wno-unused-function -Wno-nullability-completeness -Wno-expansion-to-defined -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Wno-unused-but-set-variable -Wno-unused-result -pipe -ggdb -march=native -fstack-protector-all -O2 -fdiagnostics-color=always -frecord-gcc-switches -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wformat -Werror=format-security -DH2O_ROOT="/usr" -DH2O_CONFIG_PATH="/etc/h2o/h2o.conf" -DH2O_LOCALSTATEDIR="/var" -pthread -D_GNU_SOURCE -fPIC -MD -MT CMakeFiles/libh2o.dir/lib/common/multithread.c.o -MF CMakeFiles/libh2o.dir/lib/common/multithread.c.o.d -o CMakeFiles/libh2o.dir/lib/common/multithread.c.o -c /var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/lib/common/multithread.c
/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/lib/common/multithread.c: In function ‘h2o_multithread_get_loop’:
/var/tmp/portage/www-servers/h2o-9999/work/h2o-9999/lib/common/multithread.c:240:13: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  240 |     return ((uv_handle_t *)&queue->async)->loop;
      |            ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At top level:
cc1: note: unrecognized command-line option ‘-Wno-nullability-completeness’ may have been intended to silence earlier diagnostics
cc1: some warnings being treated as errors
```

Originally reported downstream at: https://bugs.gentoo.org/967654
Full build log: [build.log](https://github.com/user-attachments/files/24424595/build.log)

--- a/lib/common/multithread.c
+++ b/lib/common/multithread.c
@@ -237,7 +237,11 @@ h2o_loop_t *h2o_multithread_get_loop(h2o_multithread_queue_t *queue)
     if (queue == NULL)
         return NULL;
 #if H2O_USE_LIBUV
+#if UV_VERSION_MAJOR > 1 || (UV_VERSION_MAJOR == 1 && UV_VERSION_MINOR >= 19)
+    return uv_handle_get_loop((uv_handle_t *)&queue->async);
+#else
     return ((uv_handle_t *)&queue->async)->loop;
+#endif
 #else
     return h2o_socket_get_loop(queue->async.read);
 #endif

